wifiphisher/common/macmatcher.py

```diff
@@ -1,133 +1,61 @@
-"""
-This module was made to match MAC address with vendors
-"""
+"""This module handles matching MAC address with vendors."""
 
-import wifiphisher.common.constants as constants
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from logging import getLogger
 
 
 class MACMatcher(object):
-    """
-    This class handles Organizationally Unique Identifiers (OUIs).
-    The original data comes from http://standards.ieee.org/regauth/
-    oui/oui.tx
+    """Handles Organizationally Unique Identifiers (OUIs).
+
+    The original data comes from
+    http://standards.ieee.org/regauth/oui/oui.txt
 
     .. seealso:: http://standards.ieee.org/faqs/OUI.html
     """
 
-    def __init__(self, mac_vendor_file):
-        """
-        Setup the class with all the given arguments
-
-        :param self: A MACMatcher object
-        :param mac_vendor_file: The path of the vendor file
-        :type self: MACMatcher
-        :type mac_vendor_file: string
-        :return: None
-        :rtype: None
-        """
-
-        self._mac_to_vendor = {}
-        self._vendor_file = mac_vendor_file
+    def __init__(self, vendor_file):
+        # type: (Text) -> None
+        """Initialize the class with all the given arguments."""
+        self.logger = getLogger(__name__)
+        self._vendors = dict()  # type: Dict[Text, Text]
+        self._vendor_file = vendor_file
 
         # get the information in the vendor file
         self._get_vendor_information()
 
     def _get_vendor_information(self):
-        """
-        Read and process all the data in the vendor file
-
-        :param self: A MACMatcher object
-        :type self: MACMatcher
-        :return: None
-        :rtype: None
-        """
-
-        # open the file with all the MAC addresses and
-        # vendor information
-        with open(self._vendor_file, 'r') as _file:
-            # check every line in the file
+        # type: () -> None
+        """Read and process all the data in the vendor file."""
+        self.logger.debug("Parsing the vendor file")
+        with open(self._vendor_file) as _file:
             for line in _file:
                 # skip comment lines
                 if not line.startswith("#"):
-                    # separate vendor and MAC addresses and add it
-                    # to the dictionary
-                    separated_line = line.rstrip('\n').split('|')
-                    mac_identifier = separated_line[0]
-                    vendor = separated_line[1]
-                    logo = separated_line[2]
-                    self._mac_to_vendor[mac_identifier] = (vendor, logo)
+                    vendor_id, vendor_name = line.rstrip('\n').split("\t")
```

Comment:
This is an important change that is not exactly counted as "refactoring". Make sure you mention these kind of changes in your PR or it may be left unnoticed.

I argue against changing the format of the entries in `wifiphisher-mac-prefixes`. The previous delimiter ("|") was concrete and allowed us to extend the format even more in the future if needed. For example, an entry in the future may be "<mac>|<vendor>|<vendor_details>|<vendor_short_name>".

Also, when you changed the entries in the file, I'm not sure they were properly altered. I see spaces but no tabs.